pallets/parachain-staking/src/lib.rs

```diff
@@ -939,13 +939,73 @@ pub mod pallet {
 		(reads, writes)
 	}
 
+	pub(crate) fn correct_max_nominations_per_collator_upgrade_mistake<T: Config>() -> (u64, u64) {
+		let (mut reads, mut writes) = (0u64, 0u64);
+		for (account, state) in <CollatorState2<T>>::iter() {
+			reads += 1u64;
+			// 1. collect all nominator amounts into single vec and order them
+			let mut all_nominators = state.top_nominators.clone();
+			let mut starting_bottom_nominators = state.bottom_nominators.clone();
+			all_nominators.append(&mut starting_bottom_nominators);
+			// sort all nominators from greatest to least
+			all_nominators.sort_unstable_by(|a, b| b.amount.cmp(&a.amount));
```

Comment:
could run on this only if `!state.bottom_nominators.is_empty()` if it is determined running this on all CollatorState is especially expensive